mongo/integration/unified/unified_spec_runner.go

```diff
@@ -114,22 +126,42 @@ func RunTestFile(t *testing.T, filepath string, opts ...*Options) {
 			CreateClient(false)
 
 		mt.RunOpts(testCase.Description, mtOpts, func(mt *mtest.T) {
-			if err := testCase.Run(mt); err != nil {
-				mt.Fatal(err)
+			defer func() {
+				// catch panics from looking up elements and fail if it's unexpected
+				if r := recover(); r != nil {
+					if pass {
+						mt.Fatal(r)
+					}
+				}
+			}()
+			err := testCase.Run(mt)
+			if pass {
+				if err != nil {
+					mt.Fatal(err)
+				}
+				return
+			}
+			if err != nil {
+				return
 			}
+			mt.Fatalf("Expected test to error, got nil")
 		})
 	}
 }
 
 // ParseTestFile create an array of TestCases from the testJSON json blob
-func ParseTestFile(t *testing.T, testJSON []byte, opts ...*Options) ([]mtest.RunOnBlock, []*TestCase) {
+func ParseTestFile(testJSON []byte, opts ...*Options) ([]mtest.RunOnBlock, []*TestCase, error) {
```

Comment:
I was trying to main the function signatures as much as possible, but there's no other way to get the error back from this.

Reply:
When would `valid/fail` expect an error from parsing? Wouldn't that imply that it's "invalid"?